src/Database/Esqueleto/PostgreSQL/JSON.hs

```diff
@@ -0,0 +1,590 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-|
+  This module contains PostgreSQL-specific JSON functions.
+
+  A couple of things to keep in mind about this module:
+
+      * The @Type@ column in the PostgreSQL documentation tables
+      are the types of the right operand, the left is always @jsonb@.
+      * This module also exports 'PersistField' and 'PersistFieldSql'
+      orphan instances for 'Data.Aeson.Value'
+      * Since these operators can all take @NULL@ values as their input,
+      and most can also output @NULL@ values (even when the inputs are
+      guaranteed to not be NULL), all 'Data.Aeson.Value's are wrapped in
+      'Maybe'. This also makes it easier to chain them.
+      Just use the 'just' function to lift any non-'Maybe' JSON values
+      in case it doesn't type check.
+      * As long as the previous operator's resulting value is
+      'Maybe' 'Data.Aeson.Value', any other JSON operator can be
+      used to transform the JSON further.
+      (e.g. @'[1,2,3]' -> 1 \@> 2@)
+
+  /The PostgreSQL version the functions work with are included/
+  /in their description./
+
+  /Since: 3.1.0/
```

Comment:
I've used the `/Since: 3.1.0/` notation, since that's what's used throughout the entire package, but I can easily fix it, if you'd prefer the `@` notation?

Reply:
Yeah I think the `@since` notation is newer than the package. It potentially has some semantic meaning to Haddock so it's probably better to add it.